File: backend/WarehouseInventory.Api/Contracts/Products/ProductDetailsDto.cs
namespace WarehouseInventory.Api.Contracts.Products;

public record ProductDetailsDto(
    Guid Id,
    string Sku,
    string? Barcode,
    string Name,
    string Category,
    string Location,
    decimal Price,
    int CurrentStock,
    int ReorderThreshold,
    bool IsActive,
    Guid Version
);

File: frontend/src/app/data/dto/product-details.dto.ts
export interface ProductDetailsDto {
  id: string;
  sku: string;
  barcode: string | null;
  name: string;
  category: string;
  location: string;
  price: number;
  currentStock: number;
  reorderThreshold: number;
  isActive: boolean;
  version: string;
}
